be/src/vec/functions/function_string.cpp

```diff
@@ -605,7 +606,7 @@ using FunctionStringLocate =
 using FunctionStringFindInSet =
         FunctionBinaryToType<DataTypeString, DataTypeString, StringFindInSetImpl, NameFindInSet>;
 
-using FunctionReverse = FunctionStringToString<ReverseImpl, NameReverse>;
+using FunctionReverse = FunctionReverseCommon<ReverseImpl, ArrayReverseImpl, NameReverse>;
```

Comment:
try to register for String and Array separately.

Reply:
since the 'reverse()' is not the variadic arguments function, it just can register one function with one name.